thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/datasource/ThirdEyeCacheRegistry.java

```diff
@@ -90,14 +98,42 @@ public static void initDataSources(ThirdEyeConfiguration thirdeyeConfig) {
     }
   }
 
+  public static void initCentralizedCache(ThirdEyeConfiguration thirdeyeConfig) {
+    try {
+      URL cacheConfigUrl = thirdeyeConfig.getCacheConfigAsUrl();
+      CacheConfig cacheConfig = CacheConfigLoader.fromCacheConfigUrl(cacheConfigUrl);
+      if (cacheConfig == null) {
+        throw new IllegalStateException("Could not get cache config from path " + cacheConfigUrl);
+      }
+
+      CacheDataSource dataSource = CacheConfig.getCentralizedCacheSettings().getDataSourceConfig();
+
+      cacheConfig.setHost(dataSource.getHost());
+      cacheConfig.setAuthUsername(dataSource.getAuthUsername());
+      cacheConfig.setAuthPassword(dataSource.getAuthPassword());
+      cacheConfig.setBucketName(dataSource.getBucketName());
+
+      if (INSTANCE.getTimeSeriesCache() == null) {
+        TimeSeriesCache timeSeriesCache = new DefaultTimeSeriesCache(DAO_REGISTRY.getMetricConfigDAO(), DAO_REGISTRY.getDatasetConfigDAO(),
+            ThirdEyeCacheRegistry.getInstance().getQueryCache(),
+            new CouchbaseCacheDAO(),
+            Executors.newFixedThreadPool(CacheConfig.getCentralizedCacheSettings().getMaxParallelInserts()));
+
+        ThirdEyeCacheRegistry.getInstance().registerTimeSeriesCache(timeSeriesCache);
+      }
+    } catch (Exception e) {
+      LOGGER.info("Caught exception while initializing centralized cache", e);
```

Comment:
exception is probably better. When we make queries to the centralized cache, the code as is throws an exception if a fatal error occurs, so no detections will complete if the connection is not properly initialized.

Reply:
changed the code so that if exception occurs, we exit (since all queries to couchbase will fail anyways), but if the issue is that the config could not be read, we default to using in-memory cache.